モーダルで検索画面を表示するように変更

yaplus opened the pull request:
## 概要

TabBarの中央にボタンを表示してモーダルで検索画面が表示されるようにしました。  
ボタンのアイコンなどのUIは仮のものです。

<img src="https://user-images.githubusercontent.com/3463901/69901024-d4047f80-13be-11ea-9e6b-16f722445096.PNG" width="186"> <img src="https://user-images.githubusercontent.com/3463901/69901025-d49d1600-13be-11ea-94b1-46e8da344216.PNG" width="186"> <img src="https://user-images.githubusercontent.com/3463901/69901026-d49d1600-13be-11ea-97d5-903e2f752d05.PNG" width="186">

## 注意点

Constraintsをコードから追加するために [SnapKit](https://github.com/SnapKit/SnapKit) を使ってます。

## やってほしいこと

動作確認とコードレビューをしていただけると助かります。

yaplus committed:
中央にボタンを追加、モーダルで検索画面を表示するように変更

yaplus committed:
UIの微調整
WelcomeViewの2回目の表示でDoneボタンを押せるようにした
MemberViewの表示を修正

yaplus committed:
MemberViewとSearchViewのNavigationBarの表示を修正

yaplus committed:
RequestViewのNavigationBarの修正

yaplus committed:
TabBarControllerのdelegateを指定し忘れてた

yaplus committed:
コメントを追記

yaplus committed:
SearchViewのNavigationItemのラベルをCancelからCloseに直したので関数名も合わせた

yaplus commented:
このプルリク通ったらとりあえず

- アプリケーションがバックグラウンドから復帰した際に再接続するようになった
- 接続しているデバイスの一覧を見られるようになった
- NowPlayingを見れるようになった（子機）
- インターフェイスを変更した

みたいなリリースノートでTestFlight版をデプロイしたい

amylaseF85 commented on DJYusaku/RequestsViewController.swift:
```diff
@@ -59,16 +59,14 @@ class RequestsViewController: UIViewController {
             // 初回にはWelcomeViewをモーダルを表示
             let storyboard: UIStoryboard = self.storyboard!
             let welcomNavigationController = storyboard.instantiateViewController(withIdentifier: "WelcomeNavigation")
             welcomNavigationController.isModalInPresentation = true
```
welcome？
typoっぽい

amylaseF85 replied:
これ実は今回の差分じゃないな

yaplus replied:
直しとく

yaplus commented:
https://github.com/enpitut2019/dj-yusaku/issues/30 のセル全体をタップしてリクエスト機能もやった方が良ければこのプルリクでやります。どう？

amylaseF85 commented on DJYusaku/TabBarController.swift:
```diff
@@ -0,0 +1,59 @@
+
+        self.delegate = self
+        
+        // 中央のタブがある位置に空のViewを配置
+        let centerView = UIView()
+        self.tabBar.addSubview(centerView)
+        centerView.snp.makeConstraints { (make) -> Void in
+            make.height.equalTo(self.tabBar.subviews[1])
+            make.width.equalTo(self.tabBar.subviews[1])
+            make.centerX.equalToSuperview()
```
俺はいつだったか、軽く説明を受けた気がするけど、
空のViewとかの存在している説明が必要な気がするけど、どうだろう
（DummyViewControllerもそれに関連してる？）

yaplus replied:
空のViewは単純に中央のTabBarItemの代わりに親Viewとして機能してるだけなんだけど、ちゃんと説明しようとするとこの実装トリック全体の説明になっちゃいそう

amylaseF85 replied:
どれくらいの詳細度にするかは申し訳ないけど任せて、3つ目のViewが実は必要で〜…という簡単な説明が欲しいなと

amylaseF85 replied:
どれくらいの詳細度にするかは申し訳ないけど任せて、3つ目のViewが実は必要で〜…という簡単な説明が欲しいなと

amylaseF85 commented on DJYusaku/TabBarController.swift:
```diff
@@ -0,0 +1,59 @@
+
+// MARK: - UITabBarControllerDelegate
+
+extension TabBarController: UITabBarControllerDelegate {
+    
+    func tabBarController(_ tabBarController: UITabBarController, shouldSelect viewController: UIViewController) -> Bool {
+        if viewController is DummyViewController { return false }   // 中央のタブは無効化しておく
+        return true
+    }
```
> 俺はいつだったか、軽く説明を受けた気がするけど、
空のViewとかの存在している説明が必要な気がするけど、どうだろう
（DummyViewControllerもそれに関連してる？）

ここで説明できてるとも言える

yaplus replied:
うーんまあそう…  
実際TabBarControllerクラス全体がトリックの説明になってると考えてる

amylaseF85 replied:
そうか…俺がちゃんとコード読めてないだけかもしれん

tsuu32 commented:
セル全体をタップしてリクエスト送るのが直感的なUIなのかどうか

amylaseF85 reviewed (commented):
2台で動作確認して、問題なさそう

bldsky commented:
> #30 のセル全体をタップしてリクエスト機能もやった方が良ければこのプルリクでやります。どう？

今のところタップして楽曲の詳細情報を確認とかないから僕はタップしてリクエストしても良いと思います

yaplus commented:
@tsuu32 @bldsky 
現状から+ボタンを取り払うと、検索結果だと曲（セル）をタップせざるを得ないようなUIだし「直感的」かどうかはわからないけど誘導としてはいいんじゃないのかなと思う（Swarmの地点選択もそんな感じだし）

yaplus committed:
typo修正

amylaseF85 commented:
検索が本来の目的である画面に+ボタンを置いてた今までならさておき、追加することを目的としているこのバージョンで、+ボタンがどうこうという話ではない気がする

じゃあなんのためにTabbarに+ボタンを置いたんだとなる

tsuu32 commented:
確かにタブにある+を押してきてるんだから、tablecellに+はいらないか

bldsky commented:
iPad mini4とiPhone7の動作確認しました。

bldsky reviewed (approved):
いいですね。うん

yaplus committed:
検索画面でセルをタップして曲を追加できるようにした
+ボタンをSearchMusicTableViewCellから削除した

yaplus commented:
セルを選択して曲追加するやつをやりました  
ついでに曲を追加した時のアラート表示を削除しました（曲が追加できたらモーダルが下がるため）

yaplus commented:
@amylaseF85 @tsuu32 @bldsky 
https://github.com/enpitut2019/dj-yusaku/pull/51/commits/98b53f1ff240addfdba273fcfb3d3a4a492a057e , https://github.com/enpitut2019/dj-yusaku/pull/51/commits/4ef6aa06245f22735750580897f5437c8ee7d6fd の内容を一応レビューしてくれ〜

yaplus committed:
NavigationBarの文言を変更

yaplus committed:
NowPlayingをMPMediaItemではなくSongでやりとりするようにした
RequestViewが表示されるときにTableViewとNowPlayingを更新するようにした（NowPlayingItemDidChangeが呼ばれないバグの影響軽減）

yaplus commented:
#23 を軽減するような変更を加えました（VIewが切り替わった時にNowPlayingItemを更新させるようにしました）

bldsky commented:
僕は確認しました

yaplus commented:
マージするでよ

yaplus commented:
しちゃった❤️

bldsky commented:
うーん

yaplus commented:
TestFlight版更新すっか

bldsky commented:
まあ全部確認した(alert消えてるのは僕もそっちの方がいいと思ったが......)

bldsky commented:
更新して

yaplus commented:
バージョン番号とビルド番号更新するプルリク挟むけどまた無断でマージするね
